build.zig

```diff
@@ -64,6 +64,187 @@ const freetypeSources = [_][]const u8{
 	"src/winfonts/winfnt.c",
 };
 
+pub fn addVulkan(b: *std.Build, c_lib: *std.Build.Step.Compile, target: std.Build.ResolvedTarget, flags: []const []const u8) void {
+	const headers = b.dependency("Vulkan-Headers", .{});
+	const loader = b.dependency("Vulkan-Loader", .{});
+
+	// NOTE(blackedout): How to compile the Vulkan loader is taken from the CMakeLists of that project.
+	// This zig build is very incomplete but somehow still works on macOS
+	const normalLoaderSources = [_][]const u8{"allocation.c", "cJSON.c", "debug_utils.c", "extension_manual.c", "loader_environment.c", "gpa_helper.c", "loader.c", "log.c", "loader_json.c", "settings.c", "terminator.c", "trampoline.c", "unknown_function_handling.c", "wsi.c"};
+
+	const win32LoaderSources = [_][]const u8{
+		"loader_windows.c",
+		"dirent_on_windows.c",
+	};
+
+	const linuxLoaderSources = [_][]const u8{
+		"loader_linux.c",
+	};
+
+	var all_flags = std.array_list.Managed([]const u8).init(b.allocator);
+	all_flags.appendSlice(flags) catch unreachable;
+	switch(target.result.os.tag) {
+		.windows => {
+			all_flags.append("-DVK_USE_PLATFORM_WIN32_KHR") catch unreachable;
+		},
+		.linux, .freebsd, .openbsd, .dragonfly => {},
+		.ios => {
+			all_flags.append("-DVK_USE_PLATFORM_METAL_EXT") catch unreachable;
+			all_flags.append("-VK_USE_PLATFORM_IOS_MVK") catch unreachable;
```

Comment:
Isn't this missing a D?
```suggestion
			all_flags.append("-DVK_USE_PLATFORM_IOS_MVK") catch unreachable;
```